WordPress/Classes/Utility/Media/MediaImageExporter.swift

```diff
@@ -57,48 +61,63 @@ class MediaImageExporter: MediaExporter {
+    ///     - onError: Called if an error was encountered during creation.
     ///
     func exportImage(_ image: UIImage, fileName: String?, onCompletion: @escaping OnImageExport, onError: @escaping OnExportError) {
         do {
-            guard let data = UIImageJPEGRepresentation(image, 1.0) else {
-                throw ImageExportError.imageJPEGDataRepresentationFailed
+            var data: Data?
+            var hint: String?
+            // If the exportImageType is targeting a PNG, try to init PNG data.
+            if let exportType = options.exportImageType, UTTypeEqual(exportType as CFString, kUTTypePNG) {
+                data = UIImagePNGRepresentation(image)
+                hint = kUTTypePNG as String
             }
-            exportImage(withJPEGData: data,
+            // If the data failed to init as PNG, or is another type, try and init as JPEG data.
+            if data == nil {
+                data = UIImageJPEGRepresentation(image, 1.0)
+                hint = kUTTypeJPEG as String
+            }
+            // Ensure that we do indeed have image data.
+            guard let imageData = data else {
+                throw ImageExportError.imageDataRepresentationFailed
+            }
+            exportImage(withData: imageData,
                         fileName: fileName,
+                        typeHint: hint,
                         onCompletion: onCompletion,
                         onError: onError)
         } catch {
             onError(exporterErrorWith(error: error))
         }
     }
 
-    /// Exports and writes an image's data, expected as JPEG format, to a local Media URL.
+    /// Exports and writes an image's data, expected as PNG or JPEG format, to a local Media URL.
     ///
-    /// - parameter fileName: Filename if it's known.
-    /// - parameter onCompletion: Called on successful export, with the local file URL of the exported UIImage.
-    /// - parameter onError: Called if an error was encountered during creation.
+    /// - Parameters:
+    ///     - fileName: Filename if it's known.
+    ///     - typeHint: Hint towards the UTType of data, such as PNG or JPEG.
+    ///     - onCompletion: Called on successful export, with the local file URL of the exported UIImage.
+    ///     - onError: Called if an error was encountered during creation.
     ///
-    func exportImage(withJPEGData data: Data, fileName: String?, onCompletion: @escaping OnImageExport, onError: @escaping OnExportError) {
+    func exportImage(withData data: Data, fileName: String?, typeHint: String?, onCompletion: @escaping OnImageExport, onError: @escaping OnExportError) {
         do {
-            let options: [String: Any] = [kCGImageSourceTypeIdentifierHint as String: kUTTypeJPEG]
-            guard let source = CGImageSourceCreateWithData(data as CFData, options as CFDictionary) else {
+            let hint = typeHint ?? kUTTypeJPEG as String
```

Comment:
Played with this a bit. Since users of this func may or may not know their type hint, I find it cleaner to just have an optional versus defining a default value and forcing two different method signatures when a hint is available or not. Such as in `func exportImage(_ image: UIImage,...)`

Maybe there's Swift trick to it, but I feel like we need a way to pass in an optional value to a parameter that has a default value, without passing `nil` if it's optional.